### ISSUE TITLE
Error in the fee calculation when uing token

### ISSUE THREAD

**Issue by chevdor** [paritytech/parity#6606]
> I'm running:
>
> - **Parity version**: master
> - **Operating system**: MacOS 
> - **And installed**:  from source

I noticed a strange behavior while trying to transfer all tokens from an account to another.
See how the fee in ETH is deducted from the total amount of token?
![image](https://user-images.githubusercontent.com/738724/30987681-2b5523ae-a498-11e7-82e6-5ae9a65136d1.png)

Trying to take over also cause troubles:
![image](https://user-images.githubusercontent.com/738724/30987704-41a48aaa-a498-11e7-9f8c-c29633288f4a.png)

**Comment by 5chdn** [comment]
Which version of Parity are you running? (master is no version)

Please indicate the exact steps how to reproduce. Probably strongly related to #5658

**Comment by chevdor** [comment]
Well it was HEAD of master at the time of the report.
I saw the issue last week, so definitely after you fixed #5658 but it sounds indeed quite related.

**Comment by 5chdn** [comment]
I'm asking because #5658 was never fixed as we were never able to reproduce it. Please help us to figure out what's causing this.

**Comment by chevdor** [comment]
Oh ok. I should be able to get back to this and test next week.

>

**Comment by chevdor** [comment]
@5chdn I wanted to test now but currently my parity does no longer show any token at all.
I will rebuild and test again, hoping I can reproduce the issue.

**Comment by chevdor** [comment]
I cannot test due to #6636

**Comment by chevdor** [comment]
Ok now that I can test again, I can tell that I still see the issue.
I made a multisig with 2 cosigners and a limit of 10.
I sent loads of tokens to it. You see only the issue if your token as decimals.
Then just send the **full account balance** checkbox.

This is NOT a UI only issue.
After signing with my 2nd cosigner, the tx went thru and I can see that the displayed value is the one considered:
https://kovan.etherscan.io/tx/0xe11e8eec331002c361f6972f7d65aa6affc62db9b9f93cf4274ea6c152644b85

After additional testing, I noticed that this is not limited to multisig. Below is a transfer from an account.

In the following example, I also boosted (a lot) the gaz price to see of the missing amount of the balance was going into fee. It does not: despite the ~6ETH fee, the seen 'total amount' is just a little under the full balance.
![image](https://user-images.githubusercontent.com/738724/31458174-877ebd3a-aebf-11e7-8499-9d4fbc41d838.png)

That makes no sense and I see the issue on a single token that was done as a demo and I wouldn´t be surprise if that token had 'funny' rules so I will put this one on my token.
